packages/root/src/navigate.ts

```diff
@@ -33,22 +26,14 @@ export default function navigation_handler(
       location,
       action,
       finish() {
-        if (
-          navigation.cancelled ||
-          pending === undefined ||
-          pending !== navigation
-        ) {
+        if (pending !== navigation) {
```

Comment:
This single check is enough to determine if the navigation should continue.

`navigation.cancelled` and `pending = undefined` are set at the same time, so checking `navigation.cancelled` is unnecessary.

When a navigation is finished/cancelled, `pending` is set to `undefined` and no longer equals `navigation`. When a navigation is superseded, `pending` is a different navigation object and no longer equals `navigation`. Combining those, if `pending !== navigation`, then we should not continue finishing/cancelling the navigation.